### PR TITLE
Fix DataFlow so it works correctly

### PR DESCRIPTION
**Pull request by JLLeitschuh**

### CODE REVIEW

File: rewrite-test/src/main/kotlin/org/openrewrite/java/dataflow/FindLocalFlowPathsStringTest.kt [hunk 1]
```diff
@@ -156,4 +156,64 @@ interface FindLocalFlowPathsStringTest: RewriteTest {
                 """
         )
     )
+
+    @Test
```
Comment (tkvangorder) [inline rewrite-test/src/main/kotlin/org/openrewrite/java/dataflow/FindLocalFlowPathsStringTest.kt:160]: Need to add the `@Disabled` to both of these tests.

Comment (tkvangorder) [reply rewrite-test/src/main/kotlin/org/openrewrite/java/dataflow/FindLocalFlowPathsStringTest.kt:160]: OK, I can assign this to Patrick.

**Comment by JLLeitschuh** [comment]
This introduces an API breaking change to DataFlow. `DataFlow::findSinks` now returns `Optional<SinkFlow<Source, Sink>>` instead of `SinkFlow<Source, Sink>` which makes it significantly easier to debug.

File: rewrite-java/src/main/java/org/openrewrite/java/dataflow/Dataflow.java [hunk 2]
```diff
@@ -25,33 +25,32 @@
...
+                return Optional.empty();
+            }
+
+            SinkFlow<Source, Sink> flow = new SinkFlow<>(start, spec);
+
+            ForwardFlow.findSinks(flow);
+
+            if (flow.isNotEmpty()) {
+                return Optional.of(flow);
             }
```
Comment (JLLeitschuh) [inline rewrite-java/src/main/java/org/openrewrite/java/dataflow/Dataflow.java:50]: A source is a source if we are currently visiting a source (or the passed cursor is a source), we don't look at parents anymore.

**Pull request merged**